Remove UnitCost and other fixes

jirrick opened the pull request:
#### Change log notes 

```
* Updated `reservations/addProduct` operation parameters, use `UnitAmount` instead of `UnitCost`.
```

#### Check during review

- [ ] JSON example extended.
  - [ ] New properties are added to correct place of the JSON.
- [ ] New properties in the table are added to correct place.  
- [ ] New operation added in the list of all operations.
- [ ] Correct formatting
  - [ ] Spacing bettween titles, sections, tables, ...
  - [ ] Correct JSON format - intedation
- [ ] DateTime properties should be allways defined like ISO format

vitasamek commented on operations/finance.md:
```diff
@@ -1093,6 +1093,19 @@ Adds new outlet bills with their items.
+                        "Currency": "USD",
+                        "NetValue": 15,
+                        "TaxCodes": [
+                            "US-DC-G"
```
Have you tested against the demo environment? All those examples should be executable against the demo. But the hotel for this access token is not in US. So the tax code will be incorrect.

bits4breakfast commented:
hi @vitasamek - sorry for jumping into this PR. This is Matteo from UpsellGuru and I'm currently going through your API documentation again for applying some change changes to the process for adding a product/item through `/api/connector/v1/orders/add` and I noticed some inconsistencies with production.

Currently I'm sending the this request (that doesn't trigger any error)
```
{"ServiceId":"adb6d971-1733-4a34-a5ed-7e2c08ea1cd9","CustomerId":"f47d2115-0743-4f95-af25-ab3a014eeb68","Items":[{"Name":"Room Upgrade (BUS => LFT)","UnitCount":1,"UnitCost":{"Amount":20,"Currency":"EUR","Tax":0},"Category":{"Code":"USG","Name":"UpsellGuru Upgrade"}}],"Notes":"Added by UpsellGuru"}
```

1. was `Category` deprecated? I can't see a deprecation notice in the changelog
2. is `UnitCost` still supported or which one should be used?
3. can I use `UnitCost` with `ProductOrders` too?
4. the response to this request includes a `ChargeId` key that is not mentioned in the documentation.

What would be the best way to get in touch with the API team?

Cheers!

jirrick commented:
Hello @bits4breakfast, it's perfectly fine to reach out to us.

#### Category
Thanks for pointing out, the `Category` parameter is active and has just been missing from documentation. The preferred way to assign category is to use `AccountingCategoryId` parameter. To get the Id you can use `accountingCategories/getAll` operation. However you still can add category by `Code` or `Name` (one is enough, `Code` has preference over `Name`) as in your example.

#### UnitCost x UnitAmount
The `UnitCost` has been deprecated as we introduced proper support for both Gross and Net pricing environments. Please use `UnitAmount` instead as support for `UnitCost` will be removed in future. In this PR I've fixed the examples to be up to date. The example for `orders/add` operation is done for Net environment, other examples are using Gross pricing environment.

#### ChargeId
This key is returned for backwards compatibility only, it has the same value as `OrderId`. You can disregard any extra information that is returned on top of what's declared in documentation.

Hope this answered your questions. If something is still not clear feel free to ask.

vitasamek commented on operations/services.md:
```diff
@@ -878,30 +878,34 @@ Creates a new order with the specified products and items. Only positive charges
-    "ClientToken": "E0D439EE522F44368DC78E1BFB03710C-D24FB11DBE31D4621C4817E028D9E1D",
-    "AccessToken": "C66EF7B239D24632943D115EDE9CB810-EA00F8FD8294692C940F6B5A8F9453D",
+   "ClientToken": "E0D439EE522F44368DC78E1BFB03710C-D24FB11DBE31D4621C4817E028D9E1D",
+    "AccessToken": "7059D2C25BF64EA681ACAB3A00B859CC-D91BFF2B1E3047A3E0DEC1D57BE1382",
```
incorrect indentation. ClientToken vs AccessToken

vitasamek commented on operations/services.md:
```diff
@@ -878,30 +878,34 @@ Creates a new order with the specified products and items. Only positive charges
-        	    "TaxCodes": [
-                    "UK-S"
-                ]
+      		    "Currency": "USD",
```
This is probably also incorrectly indented.

vitasamek commented on operations/services.md:
```diff
@@ -933,18 +937,26 @@ Creates a new order with the specified products and items. Only positive charges
 | `Currency` | string | required | ISO-4217 code of the [Currency](configuration.md#currency). |
 | `TaxCodes` | array of string [Tax Codes](configuration.md#tax-rates) | required | Tax codes to be applied to the item. (Note, you can only define one tax when sending `GrossValue`. For multiple taxes, use `NetValue`)|
 
+#### Accounting category parameters
```
This was removed in purpose. ID is the only preferred way of selection

vitasamek commented on operations/services.md:
```diff
@@ -878,30 +878,34 @@ Creates a new order with the specified products and items. Only positive charges
+		        ]
+            },
+            "AccountingCategoryId": "90eff5aa-36b4-4689-80c0-ab3a00bb412e",
+            "Category":{ 
```
Remove this

vitasamek commented:
Hi @bits4breakfast, 
thank you one more time for reaching us. As Jirka mentioned, the category is still available in the request, but only because of compatibility. The only preferred way is to use `AccountingCategoryId` as @jirrick mentioned.

vitasamek commented on operations/reservations.md:
```diff
@@ -1128,11 +1134,15 @@ Adds a new product order of the specified product to the reservation.
 | `ProductId` | string | required | Unique identifier of the [Product](services.md#product). |
 | `Count` | int | required | The amount of the products to be added. Note that if the product is charged e.g. per night, count 1 means a single product every night. Count 2 means two products every night. |
-| `UnitCost` | [Cost](services.md#cost) | optional | Unit cost of the product that overrides the cost defined in Mews. |
+| `UnitAmount` | [Amount](services.md#amount-parameters) | optional | Price of the product that overrides the price defined in Mews.  |
```
double space at the end
